common/processor/processor/src/main/java/io/helidon/common/processor/TypeInfoFactory.java

```diff
@@ -154,23 +165,36 @@ public static Optional<TypedElementInfo> createTypedElementInfoFromElement(Proce
                 .stream()
                 .map(Modifier::toString)
                 .collect(Collectors.toSet());
+        Set<TypeName> thrownChecked = Set.of();
 
-        if (v instanceof ExecutableElement) {
-            ExecutableElement ee = (ExecutableElement) v;
+        if (v instanceof ExecutableElement ee) {
             typeMirror = Objects.requireNonNull(ee.getReturnType());
             params = ee.getParameters().stream()
                     .map(it -> createTypedElementInfoFromElement(env, it, elements).orElseThrow())
-                    .collect(Collectors.toList());
+                    .toList();
             AnnotationValue annotationValue = ee.getDefaultValue();
             defaultValue = (annotationValue == null) ? null
                     : String.valueOf(annotationValue.accept(new ToAnnotationValueVisitor(elements)
-                                                     .mapBooleanToNull(true)
-                                                     .mapVoidToNull(true)
-                                                     .mapBlankArrayToNull(true)
-                                                     .mapEmptyStringToNull(true)
-                                                     .mapToSourceDeclaration(true), null));
-        } else if (v instanceof VariableElement) {
-            VariableElement ve = (VariableElement) v;
+                                                                    .mapBooleanToNull(true)
+                                                                    .mapVoidToNull(true)
+                                                                    .mapBlankArrayToNull(true)
+                                                                    .mapEmptyStringToNull(true)
+                                                                    .mapToSourceDeclaration(true), null));
+
+            thrownChecked = ee.getThrownTypes()
+                    .stream()
+                    .flatMap(it -> {
+                        if (isCheckedException(env, it)) {
+                            TypeName typeName = TypeFactory.createTypeName(it).orElse(null);
+                            if (typeName == null) {
+                                return Stream.of();
+                            }
+                            return Stream.of(typeName);
+                        }
+                        return Stream.of();
+                    })
```

Comment:
Could it simplified like this?
```suggestion
                    .filter(it -> isCheckedException(env, it))
                    .map(TypeFactory::createTypeName)
                    .flatMap(Optinal::stream)
```

Reply:
I think so, I will do it within the next PR